### PR TITLE
Added Openblock wallet support

### PR DESCRIPTION
**Pull request by Acceyuriko**

### CODE REVIEW

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Thanks for your contribution! Before we review the Openblock connector, please revert changes to `getDefaultWallets`. We have a better solution in the works to resolve the default wallet list issue, and we'd like to solely focus on connector support and stability within RainbowKit.

**Comment by Acceyuriko** [comment]
> Thanks for your contribution! Before we review the Openblock connector, please revert changes to `getDefaultWallets`. We have a better solution in the works to resolve the default wallet list issue, and we'd like to solely focus on connector support and stability within RainbowKit.

Which commit does `getDefaultWallets` refer to?

File: packages/rainbowkit/src/wallets/getDefaultWallets.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export const getDefaultWallets = ({
       wallets: [
         injectedWallet({ chains }),
         rainbowWallet({ chains }),
+        openBlockWallet({ chains }),
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/getDefaultWallets.ts:28]: @Acceyuriko Referring to this file

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

File: packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        return isAndroid() ? uri : `https://openblock.com/#/initialize/welcome`;
+      };
+
+      return {
+        connector,
+        mobile: {
+          getUri: shouldUseWalletConnect ? getUri : undefined,
+        },
+        qrCode: shouldUseWalletConnect
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts:62]: Does OpenBlock support WalletConnect? If not, should not use the `qrCode` prop

File: packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    name: 'OpenBlock',
+    iconUrl: async () => (await import('./openBlock.svg')).default,
+    iconBackground: '#fff',
+    installed: !shouldUseWalletConnect ? isOpenBlockInjected : undefined,
+    downloadUrls: {
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/openblock-bridge/memiokcjdencbponfgbkojkenpdpejhj?utm_source=chrome-ntp-icon',
+      android: 'https://openblock.com/#/initialize/welcome',
+      ios: 'https://openblock.com/#/initialize/welcome',
+      qrCode: 'https://openblock.com/#/initialize/welcome',
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts:41]: Should remove `android`, `ios`, and `qrCode` props unless there is a mobile app available

File: packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+      const getUri = async () => {
+        const { uri } = (await connector.getProvider()).connector;
+
+        return isAndroid() ? uri : `https://openblock.com/#/initialize/welcome`;
+      };
+
+      return {
+        connector,
+        mobile: {
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/openBlockWallet/openBlockWallet.ts:59]: Shouldn't use `mobile` unless an app is available

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Reference the `injectedWallet` connector implementation for an injected-only wallet connector. I tested on mobile and there doesn't appear to be a viable UX. Additionally, am seeing an issue where OpenBlock opens in a new tab on eacg dApp load. This is unexpected behavior that would likely degrade the dApp UX. Can this be corrected on the extension side?

**Comment by Acceyuriko** [comment]
If I do not set the qrCode props, the modal seems very strange. It will display the contents of the last connector I selected.
image

In addition, OpenBlock is a web wallet rather than a plug-in wallet, so when the page is opened, rainbowkit tries to automatically connect to OpenBlock, which initiates an eth_accounts request and opens the OpenBlock page.

**Comment by DanielSinclair** [comment]
1. Reference the `injectedWallet` implementation here to prevent modal issues: https://github.com/rainbow-me/rainbowkit/blob/a920c2e92dd6e48effc9b10efe61201ad044a767/packages/rainbowkit/src/wallets/walletConnectors/injectedWallet/injectedWallet.ts#LL11-L32C4
2. It's possible that there isn't an integration path for this particular kind of wallet unless the UX issues can be resolved

**Comment by Acceyuriko** [comment]
Oh, thats a pitty. Anyway, thanks for the support.

**Pull request closed without merging**